function: Migrar todos os registros do diario com o id ao invés do nome

dodocinho opened the pull request:
Função `migrarResitrosHabitos` criada para alterar os registros de todos os hábitos trocando o nome do hábito pelo id que ele possui na  collection de modelos, tanto de hábitos como de grupos de hábitos
Essa função não considera quem é o usuário daquele registro ou os hábitos já personalizados

Finishes #178378518

dodocinho committed:
feat(migrar-registros): Migração dos hábitos

Migração de todos os hábitos nos registros, substituindo os nomes pelos ids

kellynvd reviewed (changes requested):
@evandrododo na verdade os ids precisam ser os dos grupos de hábitos e hábitos dos usuários (`user/${userId}/gruposDeHabitos/${docRef.id}/habitos`) e não os dos modelos - aí vai ter que considerar qual é o usuário do registro usando o campo _userId_ . Isso porque quando os usuários editam os nomes dos habitos, eles editam os dos habitos dentro da collection _user_ e  precisamos ter os ids desses habitos ( e grupos de hábitos) no registro do diário para na hora de montar saber qual nome vamos exibir.

dodocinho committed:
fix(migrarregistros): Usando id da collection de usuario ao inves do id modelo

dodocinho committed:
fix(eslint): doublequote

dodocinho committed:
Merge branch 'develop' into feat/migrarregistros

dodocinho committed:
fix(migrarregistros): habitos inexistentes

Correção de hábitos inexistentes e troca de função assíncrona pra funções síncronas

dodocinho committed:
fix(migrarregistros): continue no lugar de return

kellynvd commented:
@evandrododo achei esses dois problemas fazendo os testes:

1- Quando já existe id no registro, ao rodar a migração ele "se perde". Exemplo:

- **Antes de rodar a migração:**

![Screenshot from 2021-07-19 14 32 51](https://user-images.githubusercontent.com/13313889/126203939-b90efd5d-2cc3-46e7-86f9-7b627afd0081.png)

Nesse caso o habito do personalizado tem um array de id. Isso vai acontecer em producao, porque já estavamos salvando o id dos habitos persoanlizados (ja que eles podem ser editados).

- **Depois da migração:**

![Screenshot from 2021-07-19 14 33 16](https://user-images.githubusercontent.com/13313889/126203995-5a090a79-bf91-45c4-b2a6-58b5b72fdf08.png)

2- Tem um habito "Companheir@" com mesmo nome dentro do grupo "Social" e dentro do grupo "Sexo". No registro eu marquei o Companheir@ do grupo Social, mas ele ficou om o id do grupo Sexo:

![image](https://user-images.githubusercontent.com/13313889/126204418-a6c2c2d2-ea4b-44f2-87fc-6840b78da140.png)
![Screenshot from 2021-07-19 14 51 39](https://user-images.githubusercontent.com/13313889/126204526-db7d35fd-2557-480a-9662-83950b6debff.png)

dodocinho committed:
fix(migrarregistros): Personalizados e duplicados

 - Não sobrescreve o id quando o registro é do grupo personalizados
 - Mapeia os hábitos considerando o grupo, evitando o caso de perder informação quando dois hábitos possuem o mesmo nome

kellynvd reviewed (approved):
tudo certo agora :tada:
